### PR TITLE
feat: user transation list page

### PR DESCRIPTION
**Pull request by matt-user**
closes FRO-15

### CODE REVIEW

File: packages/app/src/systems/Chains/eth/containers/TxListItemEthToFuel.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      );
+
+    return '';
+  }
+
+  useEffect(() => {
+    if (steps && steps[3].isDone) {
+      localStorage.setItem(`ethToFuelTx${txHash}-done`, 'true');
+    }
+  }, [steps]);
```
Comment (LuizAsFight) [inline packages/app/src/systems/Chains/eth/containers/TxListItemEthToFuel.tsx:53]: you can remove this, right ? now it's already being set inside the machine

File: packages/app/src/systems/Chains/eth/hooks/useEthAccountConnection.tsx [hunk 2]
```diff
@@ -1,37 +1,44 @@
...
-  const { data: ensAvatar } = useEnsAvatar({ address });
-  const { data: signer } = useSigner();
+  const { data: ensAvatar } = useEnsAvatar({ name: address });
   const { data: balance } = useBalance({ address });
+  const publicClient = usePublicClient();
+  const { data: walletClient } = useWalletClient();
 
   const { open: isConnecting, setOpen } = useModal();
   const { disconnect } = useDisconnect();
+  const paddedAddress = bn(address).toHex(32) as `0x${string}`;
```
Comment (LuizAsFight) [inline packages/app/src/systems/Chains/eth/hooks/useEthAccountConnection.tsx:23]: add this logic to utils `parseEthAddressToFuel`

File: packages/app/src/systems/Chains/eth/hooks/useEthDepositLogs.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  );
+
+  // TODO: remove this filtering when previous query gets fixed and returns correct data
+  const filteredLogs = query.data?.filter((log) => {
+    if (log.topics.length < 3) {
+      return false;
+    }
+
+    return (
+      // log.topics[1] === ethPaddedAddress ||
```
Comment (LuizAsFight) [inline packages/app/src/systems/Chains/eth/hooks/useEthDepositLogs.tsx:54]: remove this comment ?

File: packages/app/src/systems/Chains/eth/hooks/useTxEthToFuel.tsx [hunk 4]
```diff
@@ -65,35 +65,60 @@ const selectors = {
...
           ethTx,
           ethProvider,
           fuelProvider,
           fuelAddress,
+          ethPublicClient,
         },
       });
     }
-  }, [ethTx, ethProvider, fuelProvider, fuelAddress, service]);
+  }, [ethTx, ethProvider, fuelProvider, fuelAddress, service, ethPublicClient]);
```
Comment (LuizAsFight) [inline packages/app/src/systems/Chains/eth/hooks/useTxEthToFuel.tsx:102]: need to add `skipAnalyzeTx` and `ethPublicClient` here

File: packages/app/src/systems/Chains/eth/hooks/useTxEthToFuel.tsx [hunk 5]
```diff
@@ -65,35 +65,60 @@ const selectors = {
...
+    `ethBlockDate-${ethTx?.blockHash}`
+  );
+  const { block } = useBlock(
+    !cachedBlockDate ? (ethTx?.blockHash as `0x${string}`) : undefined
+  );
   const service = useInterpret(txEthToFuelMachine);
   const steps = useSelector(service, selectors.steps);
   useEffect(() => {
-    if (ethTx && ethProvider && fuelProvider && fuelAddress) {
+    if (ethTx && ethProvider && fuelProvider && fuelAddress && !skipAnalyzeTx) {
```
Comment (LuizAsFight) [inline packages/app/src/systems/Chains/eth/hooks/useTxEthToFuel.tsx:91]: need to add `ethPublicClient` here

File: packages/app/src/systems/Chains/eth/hooks/useTxEthToFuel.tsx [hunk 6]
```diff
@@ -65,35 +65,60 @@ const selectors = {
...
 
   return {
     handlers: {
       close: store.closeOverlay,
+      openTxEthToFuel: store.openTxEthToFuel,
     },
     ethTx,
+    ethBlockDate: cachedBlockDate
+      ? new Date(Number(cachedBlockDate))
+      : block.date || new Date(),
```
Comment (LuizAsFight) [inline packages/app/src/systems/Chains/eth/hooks/useTxEthToFuel.tsx:121]: should be `undefined` if there's also no `block.date`

```suggestion
      : block.date || undefined,
```

File: packages/app/src/systems/Chains/eth/services/txEthToFuel.ts [hunk 7]
```diff
@@ -67,24 +85,28 @@ export class TxEthToFuelService {
 
       throw e;
     }
+    return '0x';
```
Comment (LuizAsFight) [inline packages/app/src/systems/Chains/eth/services/txEthToFuel.ts:88]: why return`0x`? it should return nothing if no bridge tx happened

File: packages/app/src/systems/Chains/eth/services/txEthToFuel.ts [hunk 8]
```diff
@@ -67,24 +85,28 @@ export class TxEthToFuelService {
...
+    const { ethTx, ethPublicClient } = input;
 
+    const receipt = await ethPublicClient.waitForTransactionReceipt({
+      hash: ethTx.hash as `0x${string}`,
+    });
+    const decodedEvent = decodeEventLog({
+      abi: AbiFuelMessagePortal,
+      data: receipt.logs[0].data,
+      topics: receipt.logs[0].topics,
+    }) as any; // eslint-disable-line @typescript-eslint/no-explicit-any
```
Comment (LuizAsFight) [inline packages/app/src/systems/Chains/eth/services/txEthToFuel.ts:108]: can we try avoiding this `as any`?

**Review by LuizAsFight — APPROVED** [review]
pretty good @matt-user  🚀

changes commented can be done in next PR, as it will be the finishing one for transactions list [FRO-15](https://linear.app/fuel-network/issue/FRO-15/create-transactions-history)

File: packages/app/src/systems/Bridge/routes.tsx [hunk 9]
```diff
@@ -1,9 +1,26 @@
...
+    />
+    <Route
+      path={Pages.transactions}
+      element={
+        <BridgeHome>
+          <BridgeTxList />
+        </BridgeHome>
+      }
+    />
+  </>
```
[the remark this answers is not in the document]
Comment (matt-user) [reply packages/app/src/systems/Bridge/routes.tsx:25]: This did not work :/

**Pull request merged**